src/fasta-recs.c

```diff
@@ -9,8 +40,63 @@ int main(int argc, char const *argv[])
     }
 
     const char *fasta_fname = argv[1];
-    printf("I should be parsing %s now, but someone forgot to implement that.\n",
-           fasta_fname);
+
+    char *content = readFile(fasta_fname);
+
+    //Parsing into DNA chains
+    int pos = 0;
+    const int length = strlen(content);
+
+    // Find every chain
+    while(pos<length){
+        //First we have to find '>'
+        char c = content[pos];
+        while(pos < length && c!='>'){
+            pos++;
+            c = content[pos];
+        }
+
+        // Skip spaces
+        while(pos < length && c!=' '){
+            pos++;
+            c = content[pos];
+        }
+
+        // Find the name
+        int nameBegin = pos;
+        while(pos<length && c!=' ' && c!='\n'){
+            pos++;
+            c = content[pos];
+        }
+
+        char *name = calloc(pos-nameBegin, sizeof(char));
+        memcpy(name, &content[nameBegin], pos);
```

Comment:
Remember that strings are zero-terminated. You copy the name, but you do not terminate it. That will give you a crash later...